### PR TITLE
[TorchToLinalg] remove `extract_slice` grid_sample lowering

### PR DESCRIPTION
**Pull request by IanWood1**
Instead of using extract_slice for grid sampler, use affine constants to access the X and Y values in the generic op's region.

### CODE REVIEW

**Comment by AmosLewis** [comment]
https://github.com/iree-org/iree/pull/17692

File: lib/Conversion/TorchToLinalg/Uncategorized.cpp
Comment (zjgarvey) [inline lib/Conversion/TorchToLinalg/Uncategorized.cpp:2480]: Is it possible to have the same input twice in a linalg generic op but with different indexing maps? Or does that kind of thing cause the same problem as before?

```
gr1_indexing_map = <(d0,d1,d2,d3) -> (d0,d2,d3,1)>
gr0_indexing_map = <(d0,d1,d2,d3) -> (d0,d2,d3,0)>
```
then the ins value range would be `ValueRange{grid, grid}`?

It just seems weird to me to have tensor.extract ops inside a linalg generic, when the accessing is fairly standard.

Comment (IanWood1) [reply lib/Conversion/TorchToLinalg/Uncategorized.cpp:2480]: Both work to solve the issue. I think this might be better to do it this way vs using affine constants. `grid_sample` is very similar to `gather`, which also uses `linalg.index`. See: https://github.com/llvm/torch-mlir/blob/3225f20ab19db12e532e51fe60a9ff78b48be880/lib/Conversion/TorchToLinalg/IndirectDataMovement.cpp#L28

**Comment by vivekkhandelwal1** [comment]
Hi @IanWood1, can you please address the comment and update the PR?

File: lib/Conversion/TorchToLinalg/Uncategorized.cpp
Comment (zjgarvey) [reply lib/Conversion/TorchToLinalg/Uncategorized.cpp:2480]: Admittedly, I'm not sure how exactly iree treats generic ops, I think it will be much better to use the affine map than `tensor.extract`. The only time I would ever use `tensor.extract` in a linalg generic op payload would be if the indexing of a tensor is not affine with static striding/offsetting. E.g., something like interpolation where the extraction point needs to be computed in a more complicated way, or something like adaptive pooling where the strides and window sizes are dynamically computed from the tensor shapes.

Comment (IanWood1) [reply lib/Conversion/TorchToLinalg/Uncategorized.cpp]: I would agree that using affine maps makes more sense. That's how i first tried to solve this, so I added that commit to this branch.

**Review by zjgarvey — APPROVED** [review]

**Comment by zjgarvey** [comment]
@IanWood1 If you don't have merge permissions, let me know and I'll squash and merge this when you are ready.

**Comment by IanWood1** [comment]
@zjgarvey I just got them last week, but thank you. and thanks for the review

**Pull request merged**
